extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java

```diff
@@ -234,6 +236,33 @@ public void openshiftRequirementsNative(OpenshiftConfig openshiftConfig,
         }
     }
 
+    @BuildStep(onlyIf = { OpenshiftBuild.class })
+    public void configureExternalRegistry(ApplicationInfoBuildItem applicationInfo,
+            OpenshiftConfig openshiftConfig,
+            ContainerImageInfoBuildItem containerImageInfo,
+            BuildProducer<DecoratorBuildItem> decorator) {
+        containerImageInfo.registry.ifPresent(registry -> {
+            String imagePushSecret = openshiftConfig.imagePushSecret.orElse(applicationInfo.getName() + "-push-secret");
+            if (openshiftConfig.imagePushSecret.isPresent()) {
+                // Process
+            } else if (containerImageInfo.username.isPresent() && containerImageInfo.password.isPresent()) {
+                final String name = applicationInfo.getName() + "-push-secret";
+                decorator.produce(new DecoratorBuildItem(OPENSHIFT,
+                        new AddDockerConfigJsonSecretDecorator(name, containerImageInfo.registry.get(),
+                                containerImageInfo.username.get(), containerImageInfo.password.get())));
+            } else {
+                LOG.warn("An image registry has been configured, but no push secret or credentials.");
+                LOG.warn("Unless a push secret '" + imagePushSecret + "' is provided, the build will possibly fail!");
```

Comment:
We tend to avoid `!` in error messages.
```suggestion
                LOG.warn("Unless a push secret '" + imagePushSecret + "' is provided, the build will possibly fail.");
```